tool/src/kotlin/snapshots/diplomat_tool__kotlin__test__struct.snap

```diff
@@ -49,6 +52,51 @@ internal class MyNativeStructNative: Structure(), Structure.ByValue {
     }
 }
 
+
+internal interface Runner_DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatCallback_f: Callback {
+    fun invoke(ignored: Pointer?, arg0: Int ): Int
+}
+
+class DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatCallback_f_Native: Structure(), Structure.ByValue {
+    @JvmField
+    internal var dataa: Pointer = Pointer(0L);
+    @JvmField
+    internal var run_callback: Runner_DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatCallback_f
+        = object :  Runner_DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatCallback_f {
+                override fun invoke(ignored: Pointer?, arg0: Int ): Int {
+                    throw Exception("ERROR NOT IMPLEMENTED")
+                }
+            }
+    @JvmField
+    internal var destructor: Pointer = Pointer(0L);
+
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("dataa", "run_callback", "destructor")
+    }
+}
+
+class DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatCallback_f internal constructor (
```

Comment:
If these are supposed to be public then the combined snake/Pascal case feels unidiomatic. If, however we adapted it so this wrapper is only instantiated inside the `CallbackWrapper.testMultiArgCallback` method body, then I think we wouldn't need to care.

Reply:
Good point -- I'll make them internal, this goes along well with your comments about making the usage more ergonomic and taking the callback itself as an argument.